assignments/prototypes.js

```diff
@@ -102,9 +143,130 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+
 
   // Stretch task: 
   // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+/*
+  === Villain ===
+  Attributes:
+  * affliation: 'Villain
+
+ Prototype Functions:
+  * warCry()
+  * fireballThrow()
+*/
+function Villain (attributes) {
+  Humanoid.call(this, attributes)
+  this.affliation = 'Villain'
+}
+
+Villain.prototype = Object.create(Humanoid.prototype)
+Villain.prototype.warCry = function () {
+  return `${this.affliation}s will conquer the world!`
+}
+//Add an attack method
+Villain.prototype.fireballThrow = function (opponent) {
+  opponent.healthPoints = opponent.healthPoints - this.attackPoints
+}
+
+/*
+  === Hero ===
+  Attributes:
+  * affliation: 'Hero'
+
+ Prototype Functions:
+  * battleCry()
+  * swordStrike()
+*/
+function Hero (attributes) {
+  Humanoid.call(this, attributes)
+  this.affliation = 'Hero'
+}
+
+Hero.prototype = Object.create(Humanoid.prototype)
+Hero.prototype.battleCry = function () {
+  return `${this.affliation}es will save the world, YO JOE!!!`
```

Comment:
I can't tell if this is an Easter Egg for me or a GI Joe reference! This is probably the best game I've yet seen for this project.

Reply:
Yes that is a GI Joe, reference. I purposely mixed genres to see if anyone would notice. Haha!